service/src/main/java/bio/terra/workspace/service/workspace/model/GcpCloudContext.java

```diff
@@ -80,14 +81,29 @@ public String serialize() {
     return DbSerDes.toJson(dbContext);
   }
 
-  public static GcpCloudContext deserialize(DbCloudContext dbCloudContext) {
+  /**
+   * Deserialize a database entry into a GCP cloud context.
+   *
+   * <p>This can return an empty optional if the provided DB entry is in the CREATING state. These
+   * are placeholder entries used for state management, and do not have the context (policy groups,
+   * project ID) that callers would expect from a fully-completed GCP cloud context.
+   *
+   * @param dbCloudContext Entry from cloud_context DB describing a GCP cloud context
+   * @return Empty optional if dbCloudContext is in state CREATING. Otherwise, a deserialized
+   *     GcpCloudContext object.
+   */
+  public static Optional<GcpCloudContext> deserialize(DbCloudContext dbCloudContext) {
     // Sanity check the input
     if (dbCloudContext == null
         || dbCloudContext.getSpendProfile() == null
         || dbCloudContext.getCloudPlatform() != CloudPlatform.GCP) {
       throw new InternalLogicException("Invalid GCP cloud context state");
     }
 
+    if (dbCloudContext.getState() == WsmResourceState.CREATING) {
```

Comment:
or  broken?

Reply:
BROKEN also probably doesn't have the fields filled in. Right now, no one is using BROKEN, so it probably doesn't matter.